subsys/bootloader/include/fw_metadata.h

```diff
@@ -38,6 +38,19 @@ struct __packed fw_firmware_info {
 				   */
 };
 
+/* Static asserts to ensure compatibility */
+static_assert(offsetof(struct fw_firmware_info, magic) == 0,
```

Comment:
Looks like we should create a macro which contains the common elements. So that the only thing you need to write is 

```
MY_CHECK_MACRO(fw_firmware_info, magic);
MY_CHECK_MACRO(fw_firmware_info, firmware_size);
/* and so on */
```

Reply:
Done